Flagsmith.FlagsmithClient/FlagsmithClient.cs

```diff
@@ -96,17 +96,20 @@ public class FlagsmithClient : IFlagsmithClient
             this._httpClient = httpClient ?? new HttpClient();
             this.CacheConfig = cacheConfig ?? new CacheConfig(false);
             _engine = new Engine();
+
             if (EnableAnalytics)
                 _analyticsProcessor = new AnalyticsProcessor(this._httpClient, EnvironmentKey, ApiUrl, Logger, CustomHeaders);
+
             if (EnableClientSideEvaluation)
             {
                 _pollingManager = new PollingManager(GetAndUpdateEnvironmentFromApi, EnvironmentRefreshIntervalSeconds);
                 _ = _pollingManager.StartPoll();
             }
+
             if (CacheConfig.Enabled)
             {
                 _regularFlagListCache = new RegularFlagListCache(new DateTimeProvider(),
-                    GetEnvironmentFlags().Result,
+                    new Flags(new List<IFlag>(), _analyticsProcessor, DefaultFlagHandler),
```

Comment:
I don't really understand the logic here - can't we just add a constructor on the `RegularFlagListCache` class that doesn't include the Flags and initialise it there or as null perhaps? This feels a little bit hacky again, having to initialise an empty flags list for no real reason.

Reply:
Yeah I totally understand... The problem is that the `Flags` object requires an `analyticsProcessor` and a `FlagHandler`. These must be injected but they are only known in the FlagsmithClient...
But I will find something that is more eleguant! Give me a couple of hours :)

Reply:
Fixed!